File: packages/client/android/src/main/java/com/rnota/rollback/RollbackReason.kt
package com.rnota.rollback

/**
 * Why a rollback decision was made or skipped.
 */
enum class RollbackReason {
  NONE,
  /** [com.rnota.watchdog.BootWatchdog.requiresRollback] was false. */
  NOT_REQUIRED,
  /** Watchdog requested recovery. */
  WATCHDOG_REQUIRED,
  /** Caller invoked [RollbackEngine.rollbackToPrevious]. */
  EXPLICIT_PREVIOUS,
  /** Caller invoked [RollbackEngine.rollbackToEmbedded]. */
  EXPLICIT_EMBEDDED,
  /** `previousSlot` was null. */
  PREVIOUS_MISSING,
  /** `previousSlot` directory or `bundle.hbc` missing → embedded used instead. */
  PREVIOUS_BUNDLE_MISSING,
  /** Pointers already match the target (idempotent). */
  ALREADY_AT_TARGET,
}

File: packages/client/ios/RNOta/Rollback/RollbackReason.swift
import Foundation

public enum RollbackReason: String, Sendable, Equatable {
  case none
  case notRequired
  case watchdogRequired
  case explicitPrevious
  case explicitEmbedded
  case previousMissing
  case previousBundleMissing
  case alreadyAtTarget
}
